spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/serviceregistry/EurekaAutoServiceRegistration.java

```diff
@@ -119,11 +119,14 @@ public int getOrder() {
 	@EventListener(WebServerInitializedEvent.class)
 	public void onApplicationEvent(WebServerInitializedEvent event) {
 		// TODO: take SSL into account
-		int localPort = event.getWebServer().getPort();
-		if (this.port.get() == 0) {
-			log.info("Updating port to " + localPort);
-			this.port.compareAndSet(0, localPort);
-			start();
+		String contextName = event.getApplicationContext().getServerNamespace();
+		if (StringUtils.isEmpty(contextName)) {
```

Comment:
Is there a guarantee that this will always be true?

Reply:
No guarantee from what i can see as the web app context has a `setServerNamespace` method, but this only appears to be called by the actuator.

https://github.com/spring-projects/spring-boot/search?q=serverNamespace&unscoped_q=serverNamespace

Reply:
I'll invert the logic to filter out management server namespace

Reply:
Yeah, if a user sets it, this won't hold either. Need to check with the boot team to see if there is a better way to identify an actuator context.

Reply:
Well another way might be to replace the port instance variable:

```
@LocalServerPort
int port;
```

and then the entire `onApplicationEvent` method can be simplified:

```
@EventListener(ApplicationStartedEvent.class)
public void onApplicationEvent(ApplicationStartedEvent event) {
	start();
}
```